1245 link karma number to breakdown of karma on user profile page

Maroo-b opened the pull request:
Addresses: https://github.com/AgileVentures/WebsiteOne/issues/1245

Changes description:

* link to user profile from list of user is accessible via clicking on user full name or avatar
* add link to user activity profile tab via karma number
* add verification method that displays a flash notice if user doesn't contributed yet
* add helpers method to handle active class changes for user tabs

tansaku commented:
Thanks for this @Maroo-b - have got it working locally - nice work.

One thing is that we lose the effect of having the user profile go orange when we mouse over ... that's not the end of the world, but I thought it was a nice feature that highlighted to the user the clickability of the profile.  What do you think?

Maroo-b commented:
@tansaku the mouse over effect was related to  wrapping the user profile with a link tag, it's possible to keep the effect but it will lead to user confusion because he will think that the item is clickable.

I tried to add underline style but I opt for keeping the default style used for links, (UI consistency stuff and the user already learnt that the orange text is a link)

Maroo-b commented:
@tansaku I was thinking about removing link coloring from Karma number (keep only the pointer cursor), any feedback ?

tansaku commented:
@Maroo-b we definitely want consistency - but the whole area in your pull still seems clickable - so not so misleading to change color.

As regards removing colour from karma link I would keep that colour for consistency no?

tansaku commented:
@Maroo-b needs sync to develop too

Maroo-b commented:
@tansaku I updated the PR to sync it with develop and remove additional files added by mistake.

I added the hover effect on user info block. After testing it, I agree that keeping this effect will be better.

tansaku commented:
@Maroo-b that looks great - just what the doctor ordered!

tansaku commented on app/controllers/users_controller.rb:
```diff
@@ -108,4 +109,14 @@ def set_timezone_offset_range(filter_params)
+        flash.now[:notice] = 'User does not have activity log'
+      end
+    end
+  end
```
Just thinking you could shorten this up with 

```rb
def set_activity_tab(param)
  return unless param.present?
  @param_tab = param
   unless UserPresenter.new(@user).contributed?
     @param_tab = nil
     flash.now[:notice] = 'User does not have activity log'
  end
end
```

tansaku commented:
Added some comments for a refactoring
also I wonder why we have helper specs and controllers specs if they are not testing additional code paths that the acceptance tests don't catch?  What do you think?

Maroo-b commented:
@tansaku the controller spec is check the different cases:
* user with contributions
* user without contributions

The acceptance test is verifying the happy path only.
Do you think that I should remove the helper specs ?

Maroo-b commented:
@tansaku I added the refactor. Thanks.

tansaku commented:
@Maroo-b thanks for the explanation - I think that even with the two code paths all the use cases are covered by the cucumber tests - I might be wrong.  What I worry is that the helper and and controller tests bind too tightly to a particular implementation of the desired behaviour, making it more difficult to change in future.

Also I think the cucumber generally does a better job of exposing the high level intention.  I love good unit tests but I like them on services, models and things that might get moved into separate gems where they can really benefit from re-use.

That said I don't mean to stop this from being pulled in - it's great functionality that we really need.  I can see a couple of spelling issues - I'll try and push in a few changes soon.

Maroo-b commented:
@tansaku I deleted the helper test and reset the changes for users_controller_spec.rb. Also I added the an acceptance of visiting a user profile via karma link when the user didn't contribute.

It makes sense to follow the project style.

tansaku commented:
@Maroo-b thanks so much for this - note I corrected the syntax to link to the ticket like so `fixes #1245`
